syntax = "proto3";

package audio;

option go_package = "github.com/defenseunicorns/leapfrogai/pkg/client/audio";


message AudioRequest {
    string model = 1;

	// data is the actual contents of the audio file
    bytes data = 2;
    // Prompt
    string prompt = 3;

	float temperature = 4;

    //inputlanguage is the input language
	string inputlanguage = 5;

    //format is the output language I think
    string format = 6; 
}

message AudioResponse {
    string task = 1;
    string language = 2;
    double duration = 3;
    repeated Segment segments = 4;
    string text = 5;

    message Segment {
        int32 id = 1;
        int32 seek = 2;
        double start = 3;
        double end = 4;
        string text = 5;
        repeated int32 tokens = 6;
        double temperature = 7;
        double avg_logprob = 8;
        double compression_ratio = 9;
        double no_speech_prob = 10;
        bool transient = 11;
    }
}



service Audio {
    rpc Translate(AudioRequest) returns (AudioResponse);
    rpc Transcribe(AudioRequest) returns (AudioResponse);
}